gevent/_socket2.py

```diff
@@ -381,7 +381,7 @@ def sendto(self, *args):
         try:
             return sock.sendto(*args)
         except error as ex:
-            if ex.args[0] != EWOULDBLOCK or timeout == 0.0:
+            if ex.args[0] != EWOULDBLOCK or self.timeout == 0.0:
```

Comment:
This does appear to be a legitimate typo. _socket3.py has the same issue in its `sendto` method.